internal/mode/static/state/resolver/resolver.go

```diff
@@ -39,11 +39,18 @@ func NewServiceResolverImpl(client client.Client) *ServiceResolverImpl {
 	return &ServiceResolverImpl{client: client}
 }
 
-// Resolve resolves a Service and Port to a list of Endpoints.
-// Returns an error if the Service or Port cannot be resolved.
-func (e *ServiceResolverImpl) Resolve(ctx context.Context, svc *v1.Service, port int32) ([]Endpoint, error) {
-	if svc == nil {
-		return nil, errors.New("cannot resolve a nil Service")
+// Resolve resolves a Service's NamespacedName and ServicePort to a list of Endpoints.
+// Returns an error if the Service or ServicePort cannot be resolved.
+func (e *ServiceResolverImpl) Resolve(
+	ctx context.Context,
+	svcNsName types.NamespacedName,
+	svcPort v1.ServicePort,
+) ([]Endpoint, error) {
+	// svcNsName and svcPort are guaranteed to be valid non-empty variables from when they are passed in
```

Comment:
I wonder how useful this comment is. Do we need make a reference here to another code?

Reply:
Yea agree this isn't that useful as we have the panic statement anyways.